### PR TITLE
 Clippy Refactor 1

### PR DESCRIPTION
**Pull request by StriderDM**
## Description
Resolved errors and greatly reduced warnings generated from running the clippy tool:
• Removed redundant closures.
• Removed 'static on constants.
• Removed calls to clone for copy types.
• Change checking for len() to is_empty().
• Unwrap_or to unwrap_or_else.
• Box<> large variants for issue #316.
• Changed &String to &Str where suggested.
• Removed unecessary return keywords.
• Added #[derive(Default)] where necessary.
• Val as i64 became i64::from(Val).
• Removed unnecessary use of format!.
• Removed unecessary map where necessary.
• Changed .or() to .or_else().
• Etc.

Additionally refactored grpc_wallet to use pnet instead of local-ip as done with console_text_messenger.

## Motivation and Context
See above.

## How Has This Been Tested?
cargo test --all
cargo clippy

## Types of changes

* [x] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [x] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: base_layer/core/src/block.rs [hunk 1]
```diff
@@ -238,16 +238,11 @@ impl BlockBuilder {
...
-            match iter.next() {
-                Some(tx) => {
-                    self = self.add_inputs(tx.body.inputs);
-                    self = self.add_outputs(tx.body.outputs);
-                    self = self.add_kernels(tx.body.kernels);
-                    self.header.total_kernel_offset = self.header.total_kernel_offset + tx.offset;
-                },
-                None => break,
-            }
+        while let Some(tx) = iter.next() {
```
Comment (mikethetike) [inline base_layer/core/src/block.rs]: Any reason this can't just be a `for` loop?

File: base_layer/core/src/chain.rs [hunk 2]
```diff
@@ -119,7 +119,7 @@ impl Chain {
     {
         // The searched hash should always be in the orphan list
         unorphaned_blocks.push(block_hash.clone()); // save all orphans we have used
-        let block = self.orphans.get(block_hash).unwrap().clone(); // todo remove this clone
+        let block = &self.orphans[block_hash].clone(); // todo remove this clone
```
Comment (mikethetike) [inline base_layer/core/src/chain.rs]: This clone might not be necessary after this change

File: base_layer/core/src/block.rs (hunk 1 above)
Comment (StriderDM) [reply base_layer/core/src/block.rs]: Thanks. Rewritten as a for loop.

File: base_layer/core/src/chain.rs (hunk 2 above)
Comment (StriderDM) [reply base_layer/core/src/chain.rs]: Removing the clone results in a borrowing problem.

File: applications/grpc_wallet/src/main.rs [hunk 3]
```diff
@@ -21,6 +21,9 @@
 // USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #[macro_use]
 extern crate clap;
+extern crate pnet;
```
[the remark this answers is not in the document]
Comment (StriderDM) [reply applications/grpc_wallet/src/main.rs]: Thanks, done.

File: applications/grpc_wallet/src/main.rs [hunk 4]
```diff
@@ -189,13 +192,40 @@ pub fn main() {
...
+    let interface = interfaces.first().unwrap().clone();
+    let local_ip = interface
+        .ips
+        .iter()
+        .find(|addr| addr.is_ipv4())
+        .unwrap()
+        .ip()
+        .to_string();
+
+    // println!("{}", local_ip);
```
[the remark this answers is not in the document]
Comment (StriderDM) [reply applications/grpc_wallet/src/main.rs]: Have taken it out, thanks

File: base_layer/wallet/src/text_message_service/model.rs [hunk 5]
```diff
@@ -50,7 +50,7 @@ use tari_utilities::{byte_array::ByteArray, hex::Hex};
 pub fn generate_id<D: Digest>(
     source_pub_key: &CommsPublicKey,
     dest_pub_key: &CommsPublicKey,
-    message: &String,
+    message: &str, //&String,
```
[the remark this answers is not in the document]
Comment (StriderDM) [reply base_layer/wallet/src/text_message_service/model.rs]: Comment has been removed, thanks

File: .project [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (CjS77) [inline .project]: Please don't check this file in

You ca add it to `.git/info/exclude` so that it's locally ignored

Comment (StriderDM) [reply .project]: Apologies, have removed it.

**Pull request merged**